flashlight/autograd/Functions.cpp

```diff
@@ -236,35 +246,51 @@ Variable operator!(const Variable& input) {
   return Variable(result, false);
 }
 
-Variable max(const Variable& lhs, const Variable& rhs) {
+Variable
+max(const Variable& lhs, const Variable& rhs, const bool useMixedPrecision) {
```

Comment:
The nice thing about removing this flag and removing `autoCast` is that we don't need any autocast logic at all - we basically just do the operation in whatever precision the input Variables had, which ArrayFire will do by default.